### PR TITLE
Dissociate R2dbcData and R2dbc Dsl

### PR DESCRIPTION
**Pull request by fteychene**
When initializing `R2dbcDsl` using `r2dbc` Kofu initialized automatically R2dbcDataInitializer, forcing dependency for people to `spring-data-r2dbc`.
This PR aims to dissociate R2dbcInitializer and R2dbcDataInitializer by providing a wrapper dsl around `r2dbc` (named `dataR2dbc`) to initialize spring data specific tools.

To configure R2dbc vanilla without spring-data:
```
r2dbc {
    ...
}
```

To configure R2dbc with spring-data:
```
dataR2dbc {
    r2dbc {
        ...
    }
}
```

Fix: #314

### CODE REVIEW

**Comment by fteychene** [comment]
:warning: This PR is a draft aiming to check concept and does not update Jafu and samples

File: samples/jafu-reactive-data-r2dbc/src/test/java/com/sample/IntegrationTests.java [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+		client.get().uri("/api/user").exchange()
+				.expectStatus().is2xxSuccessful()
+				.expectHeader().contentType(MediaType.APPLICATION_JSON_VALUE)
+				.expectBodyList(User.class)
+				.hasSize(3);
+	}
+
+	@Test
+	public void requestHttpApiEndpointForGettingOneSpecificUser() {
+		client.get().uri("/api/user/blozel").exchange()
```
Comment (aneveu) [inline samples/jafu-reactive-data-r2dbc/src/test/java/com/sample/IntegrationTests.java:40]: There is a type in the URL :  `"/api/user/b**C**lozel`

File: samples/jafu-reactive-r2dbc/src/test/java/com/sample/IntegrationTests.java [hunk 2]
```diff
@@ -26,6 +26,23 @@ void requestHttpApiEndpoint() {
...
+		client.get().uri("/api/user").exchange()
+				.expectStatus().is2xxSuccessful()
+				.expectHeader().contentType(MediaType.APPLICATION_JSON_VALUE)
+				.expectBodyList(User.class)
+				.hasSize(3);
+	}
+
+	@Test
+	public void requestHttpApiEndpointForGettingOneSpecificUser() {
+		client.get().uri("/api/user/blozel").exchange()
```
Comment (aneveu) [inline samples/jafu-reactive-r2dbc/src/test/java/com/sample/IntegrationTests.java:40]: same here

File: samples/kofu-reactive-data-r2dbc/src/test/kotlin/com/sample/IntegrationTests.kt [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+		client.get().uri("/api/user").exchange()
+				.expectStatus().is2xxSuccessful
+				.expectHeader().contentType(MediaType.APPLICATION_JSON_VALUE)
+				.expectBodyList<User>()
+				.hasSize(3)
+	}
+
+	@Test
+	fun `Request HTTP API endpoint for getting one specified user`() {
+		client.get().uri("/api/user/blozel").exchange()
```
Comment (aneveu) [inline samples/kofu-reactive-data-r2dbc/src/test/kotlin/com/sample/IntegrationTests.kt:42]: and here 😄

**Review by aneveu — APPROVED** [review]

File: samples/jafu-reactive-data-r2dbc/src/test/java/com/sample/IntegrationTests.java (hunk 1 above)
Comment (fteychene) [reply samples/jafu-reactive-data-r2dbc/src/test/java/com/sample/IntegrationTests.java]: Thanks it was a mistake a copy from the existing samples. I fudn that the test wasn't complete.
I fixed it

**Pull request merged**